### DISCUSSION TITLE
How to run different UIs from different files?

### DISCUSSION THREAD

**Discussion by clueple** [zauberzeug/nicegui#459] [Q&A] [answered]
I had 2 Python files in my local drive G:/app folder - A.py & B.py

I could run A.py without any problem. 
But when I tried running B.py in command prompt (under a virtual environment),
it showed A.py 's UI.

Is there any way to run different .py like it did in Flet?
(e.g. in CMD)
(your_env)  Flet   A.py   --->  shows   A.py 's UI
(your_env)  Flet   B.py  --->  shows B.py's UI

I couldn't figure that out after reading the fastapi and uvicorn's documentation.

Many thanks.

**Comment by falkoschindler** [comment, marked as answer]
I think you need to use different ports. NiceGUI runs with port 8080 by default. If A is still running, B should complain about the port 8080 still being in use. But the browser opens nonetheless, showing A's UI.

Just use something like `ui.run(port=8081)` in B and both UIs should work independently of each other.

**Reply by clueple** [reply]
> I think you need to use different ports. NiceGUI runs with port 8080 by default. If A is still running, B should complain about the port 8080 still being in use. But the browser opens nonetheless, showing A's UI.
> 
> Just use something like `ui.run(port=8081)` in B and both UIs should work independently of each other.

Thanks @falkoschindler, let me think about how to stop the port (after pressing 'x' to close 'A.py' and let the port listen to 'B.py'.

Thanks for your info.

**Comment by clueple** [comment]
I'm not sure if this (see original link from stackoverflow below) is correct.
What it does:
1) killing the port:8080 when it is in use by whatever program
2) launch the program (e.g. run your whatever_nicegui_app.py) with default to port:8080

Let me try if it can enable 'running different nicegui_app.py files without needing to set the port other than 8080'.
Currently, even when you close nicegui_app1.py, 
when you run nicegui_app2.py, you will still see the ui of nicegui_app1.py in page 127.0.0:8080

stackoverflow's post regarding 'killing port:8080':
https://stackoverflow.com/questions/40118878/8080-port-already-taken-issue-when-trying-to-redeploy-project-from-spring-tool-s
